akd_local_auditor/src/storage/s3/test.rs

```diff
@@ -0,0 +1,368 @@
+                    if !delete_objects.is_empty() {
+                        log::info!(
+                            "Bucket {} has {} objects in it which must be delete first",
+                            bucket_name,
+                            delete_objects.len()
+                        );
+                        client
+                            .delete_objects()
+                            .bucket(bucket_name)
+                            .delete(Delete::builder().set_objects(Some(delete_objects)).build())
+                            .send()
+                            .await?;
+                    }
+                    client.delete_bucket().bucket(bucket_name).send().await?;
+                    break;
+                }
+            }
+        }
+    }
+    Ok(())
+}
+
+/// Populate a test bucket with `n` audit proofs
+async fn populate_test_storage(
+    shared_config: &SdkConfig,
+    bucket: &str,
+    n_blobs: usize,
+) -> Result<()> {
+    maybe_flush_storage(shared_config, bucket).await?;
+
+    // Build the S3 config from the shared SdkConfig
+    let config = S3ConfigBuilder::from(shared_config)
+        .retry_config(RetryConfig::disabled())
+        .build();
+    // get the S3 client
+    let client = Client::from_conf(config);
+    // create the bucket
+    client
+        .create_bucket()
+        .bucket(bucket.to_string())
+        .send()
+        .await?;
+    // Generate a block of real, verifiable audit proofs
+    let proofs = generate_audit_proofs(n_blobs)
+        .await
+        .map_err(|akd_err| anyhow::anyhow!("AKD Error generating proofs: {}", akd_err))?;
+
+    // upload each proof blob into S3
+    for AuditInformation {
+        chash,
+        phash,
+        proof,
+    } in proofs
+    {
+        // Generate the s3 compat format
+        let blobs = akd::proto::generate_audit_blobs(vec![phash, chash], proof)
+            .map_err(|err| anyhow::anyhow!("Error generating audit blob {}", err))?;
+        // Grab the blob + upload it
+        if let Some(blob) = blobs.first() {
+            let byte_stream = ByteStream::from(blob.data.clone());
+            let name = blob.name.to_string();
+            client
+                .put_object()
+                .bucket(bucket.to_string())
+                .key(name)
+                .body(byte_stream)
+                .send()
+                .await?;
+        } else {
+            assert!(
+                false,
+                "We should never generate an empty blob array, but if we do crash hard & fast!"
+            );
+        }
+    }
+    Ok(())
+}
+
+fn get_s3_storage(bucket: &str) -> S3AuditStorage {
+    let clap_args = S3ClapSettings {
+        bucket: bucket.to_string(),
+        region: TEST_REGION.to_string(),
+        endpoint: Some(TEST_ENDPOINT.to_string()),
+        access_key: Some(ACCESS_KEY.to_string()),
+        secret_key: Some(SECRET_KEY.to_string()),
+    };
+    let storage: S3AuditStorage = (&clap_args).into();
```

Comment:
Can't you just return `(&clap_args).into()` here? I _think_ Rust should be smart enough to infer the type, but maybe I'm wrong.